src/tng-hooks.src.js

```diff
@@ -120,35 +120,38 @@
 		}
 	}
 
-	function useThrottle(fn, timer) {
+	function useThrottle(fn, timer, ...guards) {
 		if(!fn) { throw new TypeError('useThrottle() requires a function argument')}
 		if(!timer) { throw new TypeError('useThrottle() requires a timer argument')}
 
 		var bucket = getCurrentBucket();
 		if (bucket) {
-			const slotIdx = bucket.nextStateSlotIdx;
+			const [slot] = useState({fn, timer, lastExecute: 0});
 
-			if (!(slotIdx in bucket.stateSlots)) {
-				const slot = [
-					fn,
-					timer,
-					0
-				];
-				bucket.stateSlots[bucket.nextStateSlotIdx++] = slot;
+			if (guards.length > 0) {
+				if (guards.length === 1 && Array.isArray(guards[0])) {
+					guards = guards[0];
+				}
 			}
 
-			return function throttleFunction(...args) {
-				const [fn, timer, lastExecution] = bucket.stateSlots[slotIdx];
-				const currentTime = Date.now();
+			if (!slot.throttledFn || guardsChanged(slot.guards, guards)) {
+				slot.guards = guards;
+				slot.throttledFn = 	function throttledFunction(...args) {
+					const {fn, timer, lastExecute} = slot;
+					const currentTime = Date.now();
 
-				if(lastExecution + timer < currentTime) {
-					try {
-						fn(...args);
-					} finally {
-						bucket.stateSlots[slotIdx][2] = currentTime;
+					if(lastExecute + timer < currentTime) {
+						try {
+							fn(...args);
```

Comment:
Change to `slot.fn.apply(this,args);` so that it uses the updated function (if updated because of the guards changing).